source/Octopus.Tentacle/Kubernetes/PodLogLineParser.cs

```diff
@@ -1,68 +1,103 @@
         }
+    }
 
-        public static PodLogParseResult Success(PodLogLine logLine)
+    public class InvalidPodLogLineParseResult : PodLogLineParseResult
+    {
+        public override LogLineType Type => LogLineType.Invalid;
+
+        public string Error { get; }
+
+        public InvalidPodLogLineParseResult(string error)
         {
-            return new PodLogParseResult(true, logLine, null);
+            Error = error;
         }
-        
-        public static PodLogParseResult Fail(string error)
+    }
+    
+    public class EndOfStreamPodLogLineParseResult : ValidPodLogLineParseResult
+    {
+        public override LogLineType Type => LogLineType.EndOfStream;
+
+        public int ExitCode { get; }
+
+        public EndOfStreamPodLogLineParseResult( PodLogLine logLine, int exitCode) : base(logLine)
         {
-            return new PodLogParseResult(false, null, error);
+            ExitCode = exitCode;
         }
     }
 
     static class PodLogLineParser
     {
-        public static PodLogParseResult ParseLine(string line)
+        public static PodLogLineParseResult ParseLine(string line)
         {
             var logParts = line.Split(new[] { '|' }, 4);
 
             if (logParts.Length != 4)
             {
-                return PodLogParseResult.Fail($"Invalid log line detected. '{line}' is not correctly pipe-delimited.");
+                return new InvalidPodLogLineParseResult($"Invalid log line detected. '{line}' is not correctly pipe-delimited.");
             }
 
             if (!int.TryParse(logParts[0], out int lineNumber))
             {
-                return PodLogParseResult.Fail($"Invalid log line detected. '{logParts[0]}' is not a valid line number.");
+                return new InvalidPodLogLineParseResult($"Invalid log line detected. '{logParts[0]}' is not a valid line number.");
             }
 
             if (!DateTimeOffset.TryParse(logParts[1], out var occurred))
             {
-                return PodLogParseResult.Fail($"Invalid log line detected. Failed to parse '{logParts[1]}' as a DateTimeOffset.");
+                return new InvalidPodLogLineParseResult($"Invalid log line detected. Failed to parse '{logParts[1]}' as a DateTimeOffset.");
             }
 
             if (!Enum.TryParse(logParts[2], true, out ProcessOutputSource source))
             {
-                return PodLogParseResult.Fail($"Invalid log line detected. '{logParts[2]}' is not a valid source.");
+                return new InvalidPodLogLineParseResult($"Invalid log line detected. '{logParts[2]}' is not a valid source.");
             }
             
             //add the new line
             var message = logParts[3];
 
-            //TODO: confirm if this line is useful
-            // if (message.StartsWith("EOS-075CD4F0-8C76-491D-BA76-0879D35E9CFE"))
-            //     source = ProcessOutputSource.Debug;
+            if (message.StartsWith("EOS-075CD4F0-8C76-491D-BA76-0879D35E9CFE"))
```

Comment:
Should/Could this be a named const so it's clear what this string represents?

Reply:
Yes it should